metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlTable.java

```diff
@@ -248,16 +258,28 @@ public void updateIcebergTable(final TableInfo tableInfo) {
         log.debug("Unlocked Iceberg table {}", tableName);
     }
 
+    private void validateTableType(final QualifiedName tableName, final Map<String, String> tableMetadata) {
+        if (!tableMetadata.isEmpty()) {
+            if (ICEBERG_TABLE_TYPE.equalsIgnoreCase(tableMetadata.get(PARAM_TABLE_TYPE))) {
+                return;
+            }
+            if (VIRTUAL_VIEW_TABLE_TYPE.equalsIgnoreCase(tableMetadata.get(PARAM_TABLE_TYPE))
+                && tableMetadata.containsKey(COMMON_VIEW)
+                && Boolean.parseBoolean(tableMetadata.get(DirectSqlTable.COMMON_VIEW))
+            ) {
+                return;
+            }
+        }
+        final String message = String.format("Originally table %s is not type of iceberg ", tableName);
```

Comment:
Message should have the correct context in terms of iceberg vs common-view.

Reply:
The validation error here is that "we can't update a table that is not an iceberg table or not a common view". I have updated the message.  
I don't think we can't tell if the table should be an iceberg table/common view but it is not so that we can give out the message separately.